src/index.ts

```diff
@@ -1,75 +1,50 @@
-export { createPrismic } from "./createPrismic"
-export { usePrismic } from "./usePrismic"
+import PrismicRichText from "./PrismicRichText/PrismicRichText.vue"
+import SliceZone from "./SliceZone/SliceZone.vue"
+
+import PrismicEmbed from "./PrismicEmbed.vue"
+import PrismicImage from "./PrismicImage.vue"
+import PrismicLink from "./PrismicLink.vue"
+import PrismicText from "./PrismicText.vue"
+
+export type { SliceZoneProps } from "./SliceZone/SliceZone.vue"
+export type { PrismicRichTextProps } from "./PrismicRichText/PrismicRichText.vue"
+
+export type { PrismicEmbedProps } from "./PrismicEmbed.vue"
+export type { PrismicImageProps } from "./PrismicImage.vue"
+export type { PrismicLinkProps } from "./PrismicLink.vue"
+export type { PrismicTextProps } from "./PrismicText.vue"
 
 export {
-	// Composables
-	usePrismicImage,
-	usePrismicLink,
-	usePrismicText,
-	usePrismicRichText,
-	// Components
+	SliceZone,
 	PrismicEmbed,
 	PrismicImage,
 	PrismicLink,
 	PrismicText,
 	PrismicRichText,
-	// Slice Zone
-	getSliceComponentProps,
+}
+
+export { getRichTextComponentProps } from "./PrismicRichText"
+export type {
+	VueRichTextSerializer,
+	RichTextComponentProps,
+} from "./PrismicRichText"
+
+export {
 	TODOSliceComponent,
 	defineSliceZoneComponents,
-	SliceZone,
-} from "./components"
+	getSliceComponentProps,
+} from "./SliceZone"
 export type {
-	// Composables
-	UsePrismicImageOptions,
-	UsePrismicLinkOptions,
-	UsePrismicTextOptions,
-	UsePrismicRichTextOptions,
-	// Components
-	PrismicEmbedProps,
-	PrismicImageProps,
-	PrismicLinkProps,
-	PrismicTextProps,
-	PrismicRichTextProps,
-	// Slice Zone
-	DefineComponentSliceComponentProps,
 	SliceComponentProps,
 	SliceComponentType,
-	SliceLikeRestV2,
-	SliceLikeGraphQL,
 	SliceLike,
+	SliceLikeGraphQL,
+	SliceLikeRestV2,
 	SliceZoneComponents,
-	SliceZoneResolver,
 	SliceZoneLike,
-	SliceZoneProps,
-} from "./components"
+} from "./SliceZone"
 
-export {
-	useAllPrismicDocumentsByIDs,
-	useAllPrismicDocumentsByUIDs,
-	useAllPrismicDocumentsByTag,
-	useAllPrismicDocumentsByEveryTag,
-	useAllPrismicDocumentsBySomeTags,
-	useAllPrismicDocumentsByType,
-	useFirstPrismicDocument,
-	usePrismicDocumentByID,
-	usePrismicDocumentByUID,
-	usePrismicDocuments,
-	usePrismicDocumentsByIDs,
-	usePrismicDocumentsByUIDs,
-	usePrismicDocumentsByTag,
-	usePrismicDocumentsByEveryTag,
-	usePrismicDocumentsBySomeTags,
-	usePrismicDocumentsByType,
-	useSinglePrismicDocument,
-	dangerouslyUseAllPrismicDocuments,
```

Comment:
I believe they were never used, especially with Nuxt because they don't integrate with Nuxt's data fetching mechanism.

I was thinking about just explaining how to create your own composable should you need to, but since TanStack Query is available in Vue also (didn't know about it), I might suggest going down the same way :)